crates/edr_solidity_tests/tests/it/invariant.rs

```diff
@@ -971,3 +977,42 @@ async fn test_no_reverts_in_counterexample() {
         }
     };
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_invariant_gas_report() {
+    let filter =
+        SolidityTestFilter::new(".*", ".*", ".*fuzz/invariant/common/InvariantTest1.t.sol");
+    let mut config = TEST_DATA_DEFAULT.config_with_mock_rpc();
+    config.generate_gas_report = true;
+    let runner = TEST_DATA_DEFAULT.runner_with_config(config).await;
+    let (test_result, _) = runner.test_collect(filter).await;
+
+    assert!(test_result.gas_report.is_some());
+
+    let gas_report = test_result.gas_report.as_ref().unwrap();
+    let invariant_breaker_report = gas_report
+        .contracts
+        .get("default/fuzz/invariant/common/InvariantTest1.t.sol:InvariantBreaker")
+        .unwrap();
+
+    let deployment_report = invariant_breaker_report.deployments.first().unwrap();
+
+    assert_eq!(deployment_report.gas, 159_061);
+    assert_eq!(deployment_report.size, 434);
+    assert_eq!(deployment_report.status, GasReportExecutionStatus::Success);
+
+    assert_eq!(invariant_breaker_report.functions.len(), 3);
+    assert!(invariant_breaker_report.functions.contains_key("flag1()"));
+    assert!(invariant_breaker_report
+        .functions
+        .contains_key("set0(int256)"));
+    assert!(invariant_breaker_report
+        .functions
+        .contains_key("set1(int256)"));
+
+    let flag1_reports = invariant_breaker_report.functions.get("flag1()").unwrap();
+    assert!(!flag1_reports.is_empty());
+    assert!(flag1_reports
+        .iter()
+        .any(|r| r.gas > 0 && r.status == GasReportExecutionStatus::Success));
```

Comment:
Question: Similar to my other question, are we only interested in one case that reports gas usage? What about `Revert` and `Halt`?